src/main/java/com/czertainly/core/service/impl/DiscoveryServiceImpl.java

```diff
@@ -456,4 +484,101 @@ public List<SearchFieldDataByGroupDto> getSearchableFieldInformationByGroup() {
         logger.debug("Searchable Fields by Groups: {}", searchFieldDataByGroupDtos);
         return searchFieldDataByGroupDtos;
     }
+
+    @Override
+    public void evaluateDiscoveryTriggers(UUID discoveryUuid) {
+
+        // Get discovered certificates
+        List<DiscoveryCertificate> discoveredCertificates = discoveryCertificateRepository.findByDiscoveryUuid(discoveryUuid);
+        DiscoveryHistory discoveryHistory = discoveryRepository.findByUuid(discoveryUuid).orElse(null);
+        // Get triggers for the discovery, separately for triggers with ignore action, the rest of triggers are in given order
+        List<RuleTrigger2Object> ruleTrigger2Objects = object2TriggerRepository.findAllByResourceAndObjectUuid(Resource.DISCOVERY, discoveryUuid);
```

Comment:
select from DB could be already ordered so there would not be a need for sorting afterwards